cloudinit/sources/azure/errors.py

```diff
@@ -99,6 +107,37 @@ def __init__(self, duration: float, interface: Optional[str]) -> None:
         self.supporting_data["interface"] = interface
 
 
+class ReportableErrorImdsUrlError(ReportableError):
+    def __init__(self, *, exception: UrlError, duration: float) -> None:
+        # ConnectTimeout sub-classes ConnectError so order is important.
+        if isinstance(exception.cause, requests.ConnectTimeout):
+            reason = "connection timeout querying IMDS"
+        elif isinstance(exception.cause, requests.ConnectionError):
+            reason = "connection error querying IMDS"
+        elif isinstance(exception.cause, requests.ReadTimeout):
+            reason = "read timeout querying IMDS"
+        elif exception.code:
```

Comment:
Do you want to add other known errors such as connection reset, no route to host, network unreachable ?

Reply:
I didn't want to resort to string matching here, though we could.  The original exception is captured in supporting_data so it will be evident (those should all be a form of connection errors).